### ISSUE TITLE
Docker container fails to download GeoLite2 db file

### ISSUE THREAD

**Issue by Aryess** [shlinkio/shlink#1128]
Hi :)

#### How Shlink is set-up

* Shlink Version: 2.7.1
* PHP Version: 8.0.6
* How do you serve Shlink: Docker image

#### Summary

Testing shlink locally using the following command:
```
~|⇒ docker run \                                                        
    --name my_shlink  \
    -p 8080:8080 \
    -e SHORT_DOMAIN_HOST=test.local \
    -e SHORT_DOMAIN_SCHEMA=http \
    -e GEOLITE_LICENSE_KEY=<my key> \                                                                                                                  
    shlinkio/shlink:stable
```

#### Current behavior

Getting the following output:
```
Creating fresh database if needed...
Updating database...
Generating proxies...
Clearing entities cache...
Downloading GeoLite2 db file...

Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 36864 bytes) in /etc/shlink/vendor/shlinkio/shlink-ip-geolocation/src/GeoLite2/DbUpdater.php on line 68

Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 36864 bytes) in Unknown on line 0
```

Removing the `GEOLITE_LICENCE_KEY` line everything else works as expected, but no geoIP info is available (duh).

#### Expected behavior

GeoLite2 db should be able to be downloaded

#### How to reproduce

Running the command above fails consistently. For the record (even if it seems to be more of a php config thing), I do still have 10+ Gb of ram/disk free

**Comment by Aryess** [comment]
So, I did a bit of debugging on this, and here's what I've found:

- Downloaded the file manually from https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-City&license_key={license_key}&suffix=tar.gz (~**37Mb**)
- Extracted it (~**77Mb**)

So already we're at ~114Mb that needs to be handled by the code. Using `ini_get('memory_limit')` and `memory_get_usage(true)` ([with a bit of formatting](https://www.php.net/manual/en/function.memory-get-usage.php#96280)) just before the crash on L68 of https://github.com/shlinkio/shlink-ip-geolocation/blob/277035938d842d33d7103c11cc378ce5b5ecd709/src/GeoLite2/DbUpdater.php#L68 I get the following: `Memory: 10 mb/128M`

So that means that we only have ~118MB to handle the ~114Mb of data, leaving about ~4Mb of headspace to PharData to do its thing. Don't know how much much it's generating in the process, but for that much data I'd imagine it might need a bit more spare.

I added a `ini_set('memory_limit','256M'); ` at the beginning of that function, and the same memory debug right after the `$phar = new PharData($compressedFile);` line, and now the function is working with the following memory usage:
```
Memory before: 10 mb/256M
Memory after: 18.5 mb/256M
```

So it seems like it did indeed need a few more MB's.
@acelaya there's multiple ways to go about this, I feel like an `init_set()` in that function is a bit gross, but it's either that, bump the overall memory limit, or set the new temporary limit higher in the stack when we download the file. I'll leave it up to you to decide.

Happy to do a PR once you've decided, I'd just rather let you call the shots on that one. Thanks

**Comment by acelaya** [comment]
Thanks for the investigation @Aryess.

I haven't seen this in the past, but could be the size of the GroLite file has recently increased.

The image includes some ini config. I'll probably set the memory value there.

**Comment by acelaya** [comment]
I have just published v2.7.2, which fixes this issue by increasing the memory limit to 256M in the docker image.

Once this job has finished, the image should be available under `2.7.2` and `stable` docker tags https://github.com/shlinkio/shlink/actions/runs/1082871440

**Comment by Aryess** [comment]
Fix confirmed, everything working fine now. Thanks!

Also for posterity as I had forgotten to put it in my previous comment, I used `memory_get_peak_usage(true)` at the end of the function to see how much actually did get used, gave me 136.56 Mb. So the new limit of 256M should be good for a while.
